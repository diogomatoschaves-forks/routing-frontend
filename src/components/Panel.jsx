import React, { Component } from 'react'
import styled from 'styled-components'

const PanelWrapper = styled.div`
  position: absolute;
  z-index: 1000;
  width: 40%;
  height: 90%;
  left: 50px;
  top: 5%;
  background: rgba(30, 30, 30, 0.8);
  border-radius: 6px;
`

export default class Panel extends Component {
  render() {
    return <PanelWrapper />
  }
}
