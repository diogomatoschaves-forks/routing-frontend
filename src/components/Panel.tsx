import React, { Component } from 'react'
import styled from 'styled-components'
import { Popup, Divider } from 'semantic-ui-react'
import { Box, StyledIcon, ColoredDiv, EmptySpace } from '../styledComponents'
import InputRow from './InputRow'
import { PETROL_4, PETROL_5, PETROL_2, PETROL_1, MAIN_GREY } from '../utils/colours'
import { UpdatePoint } from '../types'


const PanelWrapper: any = styled.div`
  position: absolute;
  z-index: 1000;
  width: 32%;
  min-width: 400px;
  max-width: 600px;
  max-height: 1000px;
  /* height: 250px; */
  left: 40px;
  top: 40px;
  padding: 25px;
  background: rgba(255, 255, 255, 0.9);
  /* border: 1px solid rgb(205, 205, 205); */
  border-radius: 7px;
  box-shadow: 10px 10px 16px -9px rgba(77,77,77,0.4);
`

const StyledDivider = styled(Divider)`

  &.ui.divider {
    margin-top: 0.6rem;
    border-top: 1px solid rgb(0, 0, 0)
  }

  width: calc(100%);
`

const Panel: any = ({ updatePoint } : { updatePoint: UpdatePoint }) => {
  return (
    <PanelWrapper >
      <Box direction="column" justify="flex-start">
        <Box direction="row" justify="flex-start" padding="5px 0 15px 0">
          <Popup
            trigger={
              <ColoredDiv diameter="38" color={MAIN_GREY} margin="0 7px 0 0" position="relative">
                <Box height="100%">
                  <StyledIcon 
                    padding="0 0 0 0" 
                    overridecolor={PETROL_4} 
                    name="car"
                    size="large"
                    position="absolute"
                  />
                </Box>
              </ColoredDiv>}
            content="Car"
            position='top center'
            inverted
            style={{ borderRadius: '7px'}}
          />
          <Popup
            trigger={
              <EmptySpace width="38px" height="38px" position="relative">
                <Box height="100%">
                  <StyledIcon 
                    padding="0 0 0 0" 
                    overridecolor={PETROL_1} 
                    name="male"
                    size="large"
                    position="absolute"
                  />
                </Box>
              </EmptySpace>}
            content="Foot"
            position='top center'
            inverted
            style={{ borderRadius: '7px'}}
          />
        </Box>
        <StyledDivider />
        <InputRow 
          rowKey={'start'}
          placeholder={'Origin'}
          iconName="map marker alternate" 
          updatePoint={updatePoint}
        />
        <InputRow 
          rowKey={'end'}
          placeholder={'Destination'}
          iconName="flag checkered" 
          updatePoint={updatePoint}
        />
      </Box>
    </PanelWrapper>
  )
}

export default Panel
