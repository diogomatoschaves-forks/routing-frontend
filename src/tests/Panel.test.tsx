import React from 'react'
import TestRenderer from 'react-test-renderer'
import Panel from '../components/Panel'

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  GeolocateControl: jest.fn(),
  Map: jest.fn(() => ({
    addControl: jest.fn(),
    on: jest.fn(),
    remove: jest.fn()
  })),
  NavigationControl: jest.fn()
}));

const mockLocations = [{ 
  name: 'start', 
  marker: 'map marker alternate',
  placeholder: 'Origin',
  point: null
}, { 
  name: 'end', 
  marker: 'flag checkered',
  placeholder: 'Destination',
  point: null
}]

it('expect to render Panel component', () => {

  const testInstance = TestRenderer.create(<Panel locations={mockLocations}/>)
  expect(testInstance.toJSON()).toMatchSnapshot()
}) 
