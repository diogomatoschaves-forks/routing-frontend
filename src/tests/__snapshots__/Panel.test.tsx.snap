// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Panel component 1`] = `
<div
  className="sc-htoDjs eDjBPZ"
>
  <div
    className="sc-bdVaJa jsucjq"
    direction="column"
  >
    <div
      className="sc-bdVaJa fuiAsE"
      direction="row"
    >
      <div
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bwzfXH ksYGh"
          color={false}
          cursor="pointer"
        >
          <div
            className="sc-bdVaJa dbHwHO"
            height="100%"
          >
            <i
              aria-hidden="true"
              className="car icon sc-htpNat bxTmCt"
              fontSize="25px"
              height="20px"
              onClick={[Function]}
              overridecolor="#a6cad8"
              padding="0 0 0 0"
              position="absolute"
            />
          </div>
        </div>
      </div>
      <div
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="sc-bwzfXH ksYGh"
          color={false}
          cursor="pointer"
        >
          <div
            className="sc-bdVaJa dbHwHO"
            height="100%"
          >
            <i
              aria-hidden="true"
              className="male icon sc-htpNat bxTmCt"
              fontSize="25px"
              height="20px"
              onClick={[Function]}
              overridecolor="#a6cad8"
              padding="0 0 0 0"
              position="absolute"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ui divider sc-dnqmqq hWpGx"
    />
    <div
      className="sc-bdVaJa gSlpgJ"
      direction="row"
    >
      <div
        className="sc-bwzfXH jWeZli"
        color="#007a93"
      >
        <div
          className="sc-bdVaJa lajepV"
          height="100%"
        >
          <i
            aria-hidden="true"
            className="map marker alternate icon sc-htpNat iIqnlv"
            fontSize="25px"
            height="20px"
            onClick={[Function]}
            overridecolor="white"
            padding="0 0 0 0"
            position="absolute"
          />
        </div>
      </div>
      <div
        className="ui fluid input sc-EHOje jROHdq"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Origin"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="sc-bdVaJa gSlpgJ"
      direction="row"
    >
      <div
        className="sc-bwzfXH jWeZli"
        color="#007a93"
      >
        <div
          className="sc-bdVaJa lajepV"
          height="100%"
        >
          <i
            aria-hidden="true"
            className="flag checkered icon sc-htpNat iIqnlv"
            fontSize="25px"
            height="20px"
            onClick={[Function]}
            overridecolor="white"
            padding="0 0 0 0"
            position="absolute"
          />
        </div>
      </div>
      <div
        className="ui fluid input sc-EHOje jROHdq"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Destination"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="ui divider sc-dnqmqq hWpGx"
    />
    <div
      className="sc-bdVaJa bSEcKq"
      direction="row"
    >
      <div
        className="sc-bdVaJa jKBHhj"
        direction="column"
        width="50%"
      >
        <div
          className="sc-bdVaJa bouXiG"
          direction="row"
          height="50px"
        >
          <div
            className="ui fitted toggle checkbox custom-class sc-gZMcBi iQoUkb"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label />
          </div>
          <label
            className="sc-iwsKbI gfrWkS"
          >
            Covered Areas
          </label>
        </div>
        <div
          className="sc-bdVaJa bouXiG"
          direction="row"
          height="50px"
        >
          <div
            className="ui fitted toggle checkbox sc-gZMcBi iQoUkb"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label />
          </div>
          <label
            className="sc-iwsKbI gfrWkS"
          >
            Routing Graph
          </label>
        </div>
      </div>
      <div
        className="sc-bdVaJa fiugTd"
        direction="column"
        width="40%"
      >
        <div
          className="sc-bxivhb edpdVn"
          width="40%"
        />
      </div>
    </div>
  </div>
</div>
`;
